### ISSUE TITLE
Make periphery return non-zero exit code on warnings

### ISSUE THREAD

**Issue by Dschee** [peripheryapp/periphery#22]
I'd like to configure Periphery on our CI and configured everything like this (see our [template project](https://github.com/JamitLabs/NewProjectTemplate-iOS)):

```yaml
project: NewProjectTemplate.xcodeproj
schemes: [App]
targets: [App, Tests, UITests]
report_exclude:
  - App/Generated/SwiftGen/*.swift
  - App/SupportingFiles/*.swift
retain_public: false
quiet: true
```

But when I run `periphery scan` on the CI the builds always succeed even when I willingly add some unused code to test if the CI is failing. This is due to there being no option to make periphery return with a non-zero exit code for warnings (I don't know if it even returns with 0 on errors) – but for our purpose it would be grate to have an option like `fail_on_warnings` or `-f` or `-w` when calling the executable so this works.

**Comment by Dschee** [comment]
I just posted a PR with a basic implementation for this. Feel free to give feedback.

**Comment by Dschee** [comment]
I just removed the original periphery on my device

```bash
brew remove periphery -f
```

and installed my fork using [Mint](https://github.com/yonaskolb/Mint) (can take a while)

```bash
mint install Dschee/periphery@fail-on-warnings
```

and tested my implementation adding this to the configuration file

```yaml
fail_on_warnings: true
```

and it worked just as expected! 🎉

**Comment by ileitch** [comment]
> I don't know if it even returns with 0 on errors

It does, Commandant handles that.

**Comment by Dschee** [comment]
Implemented via #23.
